app-agent/src/exec_process.rs

```diff
@@ -17,10 +17,73 @@ use anyhow::Context;
 /// Spawns a child process and waits for it to exit.
 pub fn exec_child(external_command: String, args: Vec<String>) -> anyhow::Result<ExitStatus> {
     // Spawn the process.
-    let mut p = std::process::Command::new(external_command.clone())
-        .args(args)
-        .spawn()
-        .expect("error in child process");
+    let p_result = std::process::Command::new(external_command.clone())
+        .args(args.clone())
+        .spawn();
+
+    let mut p = match p_result {
+        Ok(val) => val,
+        Err(e) => match e.kind() {
+            std::io::ErrorKind::NotFound => {
+                let directory_entries_iter = fs::read_dir("./").unwrap();
+                let mut entry_list = String::from("Available files in the current directory:\n");
```

Comment:
I don't think it's up to the agent to replace `ls`.
How about just doing these two things:

## If `ErrorKind::NotFound`

- check if a file with the same name exists in the current directory
- print:
```
command 'XYZ' not found
Hint: A file named 'XYZ' exists in the current directory. Prepend ./ to execute it.
Example: CMD exec './XYZ'
```
where `CMD` is the alumet binary (in the rapl plugin you can find an example of how to get the path to the current alumet binary on the user).

### Bonus

If you want to have fun, implement a fuzzy finder to fix the potential typos :)
Example of what it would look like:
```sh
> ls
script.sh

> alumet-local-agent exec ./scrit.sh
ERROR: ...
Hint: did you mean './script.sh'?
```

Other example:
```sh
> ls
script.sh

> alumet-local-agent exec ./script
ERROR: ...
Hint: did you mean './script.sh'?
```

## If permission denied
- check if the file exists but does not have the X or R permission
- print:
```
file 'XYZ' is missing the following permissions: PERM
Hint: try chmod +PERM 'XYZ'
```
where `PERM` is `r`, `x` or `rx`